src/transformers/pytorch_utils.py

```diff
@@ -297,4 +297,5 @@ def id_tensor_storage(tensor: torch.Tensor) -> Tuple[torch.device, int, int]:
     else:
         unique_id = storage_ptr(tensor)
 
-    return tensor.device, unique_id, storage_size(tensor)
+    # We need to add storage_offset here as views of a tensor may share the same storage ptr.
+    return tensor.device, unique_id, storage_size(tensor), tensor.storage_offset()
```

Comment:
Shouldn't we just change `unique_id = storage_ptr(tensor) + tensor.storage_offset()` or better yet, change `storage_ptr` ?

Not sure how it plays out in TF for instance.

Reply:
@Narsil What would be the point? Using `unique_id = storage_ptr(tensor) + tensor.storage_offset()` may break in this case for example:

```
a.storage_ptr = 5
a.storage_offset = 3

b.storage_ptr = 4
b.storage_offset = 4
```

Reply:
I did not change `storage_ptr` in `safetensors` because I was thinking it was just a wrapper around `data_ptr()`. But happy to change it if you prefer (it would return a tuple though - probably breaking change).